engine/input/src/input.cpp

```diff
@@ -142,44 +142,93 @@ namespace dmInput
         return config;
     }
 
-    static GamepadBinding* NewGamepadBinding(HBinding binding, uint32_t gamepad_index)
+    static GamepadConfig* GetGamepadConfig(HBinding binding, dmHID::HGamepad gamepad, char device_name_out[dmHID::MAX_GAMEPAD_NAME_LENGTH])
     {
-        dmHID::HGamepad gamepad = dmHID::GetGamepad(binding->m_Context->m_HidContext, gamepad_index);
-        char device_name[128];
-        dmHID::GetGamepadDeviceName(binding->m_Context->m_HidContext, gamepad, device_name, sizeof(device_name));
-        if (strlen(device_name) == 0)
-        {
-            /*
-             * NOTE: We used to log a warning here but the warning is removed for the following reasons:
-             *  - The input-binding file covers several platforms and certain platforms
-             *    doesn't have support for e.g. pads. But more importantly, sometimes you might have
-             *    a device connected but sometimes not. It should be up to the user and we shouldn't
-             *    spam out warnings in such cases. In other words. It's impossible to tell whether the
-             *    warning is appropriate or not.
-             *  - We should also have support dynamic pad-connections. Whether a pad is connected
-             *    or not should be up to the game-ui.
-             */
+        char device_names[dmHID::MAX_GAMEPAD_NAME_COUNT][dmHID::MAX_GAMEPAD_NAME_LENGTH];
+        uint32_t num_names = dmHID::GetGamepadDeviceNames(binding->m_Context->m_HidContext, gamepad, device_names);
```

Comment:
I wonder if it would be enough to simply add the extra name in input.cpp?

```c++
char device_name[dmHID::MAX_GAMEPAD_NAME_LENGTH];
dmHID::GetGamepadDeviceName(binding->m_Context->m_HidContext, gamepad, device_name, sizeof(device_name));

uint32_t num_names = 1;
char** device_names[2] = { device_name, 0};

#if defined(_WIN32)
    device_names[num_names++] = "XBox 360 Controller";
#endif
```

Reply:
The "#if defined(_WIN32)" condition may ofc be different, but the overall question is the same.

Reply:
Yeah I think that's probably a good idea, I'll try it out :)